contents/testing/security-testing.md

```diff
@@ -0,0 +1,144 @@
+
+### Step 2: During Definition and Design
+
+#### 2.1 Review Security Requirements
+
+Security requirements define how an application works from a security perspective. It is essential that the security requirements are tested. Testing in this case means testing the assumptions that are made in the requirements and testing to see if there are gaps in the requirements definitions.
+
+For example, if there is a security requirement that states that users must be registered before they can get access to a particular section of a website, does this mean that the user must be registered with the system or should the user be authenticated? Ensure that requirements are as unambiguous as possible.
+
+#### 2.1 Review Design & Architecture
+
+Applications should have a documented design and architecture. This documentation can include models, textual documents, and other similar artifacts. It is essential to test these artifacts to ensure that the design and architecture enforce the appropriate level of security as defined in the requirements.
+
+Identifying security flaws in the design phase is not only one of the most cost-efficient places to identify flaws, but can be one of the most effective places to make changes. For example, if it is identified that the design calls for authorization decisions to be made in multiple places, it may be appropriate to consider a central authorization component. If the application is performing data validation at multiple places, it may be appropriate to develop a central validation framework (ie, fixing input validation in one place, rather than in hundreds of places, is far cheaper).
+
+### Step 3: During Development
+
+#### 3.1 Unit & System Tests
+
+System and unit tests should be written to explicitly ensure adherence to functional security requirements. For example, an application with an access control feature should include unit tests to ensure that unauthorised access credentials are not accepted.
+
+<box id="info">:fas-info-circle: If you would like to learn more about writing tests, you can read more [here](https://se-education.org/se-book/testing/).</box>
+
+#### 3.2 Code Walk Through
+
+The security team should perform a code walk through with the developers, and in some cases, the system architects. A code walk through is a high-level walk through of the code where the developers can explain the logic and flow of the implemented code. It allows the code review team to obtain a general understanding of the code, and allows the developers to explain why certain things were developed the way they were.
+
+The purpose is not to perform a code review, but to understand at a high level the flow, the layout, and the structure of the code that makes up the application.
```

Comment:
...understand at a high level on the flow, layout and structure of the code

Reply:
Maybe if I use a colon instead "...understand at a high level: the flow, the layout, and the structure..."?

Reply:
yep sounds good